src/frontend/app/src/components/chatrooms/RoomsList.vue

```diff
@@ -0,0 +1,211 @@
+        @click="joinRoom"
+      >
+        Join room
+      </button>
+    </div>
+  </div>
+  <div v-if="isJoining">
+    <Chatroom :chat="selectedChat" @roomLeft="setWaiting" />
+  </div>
+  <div v-if="isCreating">
+    <CreateRoom @roomCreated="setWaiting" />
+  </div>
+</template>
+
+<script lang="ts">
+import CreateRoom from "./CreateRoom.vue";
+import Chatroom from "./ChatRoom.vue";
+import { defineComponent } from "vue";
+import { makeApiCall } from "@/utils/ApiCall";
+import { Chat, AllChats } from "./Chatrooms.types.ts";
+import { useSocketStore } from "@/stores/SocketStore";
+import { mapState } from "pinia";
+import ChatLockIcon from "../icons/IconChatLock.vue";
+import ChatPrivateIcon from "../icons/IconChatPrivate.vue";
+import EyeOpen from "../icons/IconEyeOpen.vue";
+import EyeClosed from "../icons/IconEyeClosed.vue";
+
+enum State {
+  WAITING,
+  CREATING,
+  JOINING,
+}
+
+interface DataObject {
+  state: State;
+  chats: AllChats;
+  selectedChat: Chat;
+  selectedChatName: string;
+  typedPassword: string;
+  showPassword: boolean;
+  passwordIcon: any;
+}
+
+export default defineComponent({
+  data(): DataObject {
+    return {
+      state: State.WAITING,
+      chats: [],
+      selectedChat: [],
+      selectedChatName: "",
+      typedPassword: "",
+      showPassword: false,
+    };
+  },
+  computed: {
+    isWaiting() {
+      return this.state === State.WAITING;
+    },
+    isCreating() {
+      return this.state === State.CREATING;
+    },
+    isJoining() {
+      return this.state === State.JOINING;
+    },
+    passwordVisibility() {
+      return this.showPassword ? "text" : "password";
+    },
+    ...mapState(useSocketStore, {
+      socket: "chat",
+    }),
+  },
+  methods: {
+    createRoom() {
+      this.state = State.CREATING;
+    },
+    async joinRoom() {
+      this.socket.emit("joinRoom", {
+        roomName: this.selectedChatName,
+        password: this.typedPassword,
+      });
+    },
+    toggleShowPassword() {
+      this.showPassword = !this.showPassword;
+    },
+    setWaiting() {
+      this.getAllChats();
+      this.state = State.WAITING;
+    },
+    async getAllChats() {
+      const response = await makeApiCall("/chat");
+      if (response.ok) {
+        this.chats = await response.json();
+      }
+    },
+    joinRoomSuccessfully() {
+      this.selectedChat = this.findChatByName(this.selectedChatName);
+      this.state = State.JOINING;
+      this.typedPassword = "";
+      this.selectedChatName = "";
+    },
+    joinRoomFailed() {
+      console.log("Joining failed");
+    },
+    findChatByName(name: string): Chat {
+      for (let chat of this.chats.joinedChats) {
+        if (chat.name === name) {
+          return chat;
+        }
+      }
+      for (let chat of this.chats.otherChats) {
+        if (chat.name === name) {
+          return chat;
+        }
+      }
+    },
+    async refreshChatList() {
+      await this.getAllChats();
+    },
+  },
+  created() {
```

Comment:
change to mounted and removeListener on unmounted

Reply:
Mounted en unmounted toegevoegd. Cleanere code nu.